source/opt/loop_descriptor.h

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LIBSPIRV_OPT_LOOP_DESCRIPTORS_H_
+#define LIBSPIRV_OPT_LOOP_DESCRIPTORS_H_
+
+#include <cstdint>
+#include <map>
+
+#include "module.h"
+#include "pass.h"
+
+namespace spvtools {
+namespace opt {
+
+// A class to represent a loop.
+class Loop {
+ public:
+  Loop(bool is_nested, const ir::BasicBlock* begin,
+       const ir::BasicBlock* continue_target,
+       const ir::BasicBlock* merge_target)
+      : loop_start_(begin),
+        loop_continue_(continue_target),
+        loop_merge_(merge_target),
+        is_nested_(is_nested){};
```

Comment:
`const ir::BasicBlock` -> `ir::BasicBlock` (we want to operate on that loop).
This could be templated to create const/non-const Loop struct.

remove `;`

`is_nested` is unhelpful, Loop is a tree structure with a parent loop (if any) and children (a loop can have 1 to n loop)